src/HtmlUtils.tsx

```diff
@@ -499,7 +497,8 @@ export function bodyToHtml(content: IContent, highlights: Optional<string[]>, op
                 // by an attempt to search for 'foobar'.  Then again, the search query probably wouldn't work either
                 // XXX: hacky bodge to temporarily apply a textFilter to the sanitizeParams structure.
                 sanitizeParams.textFilter = function(safeText) {
-                    return highlighter.applyHighlights(safeText, safeHighlights).join('');
+                    return highlighter.applyHighlights(safeText, safeHighlights).join('')
+                        .replace(/:[\w+-]+:/g, m => opts.emotes[m] ? opts.emotes[m] : m);
```

Comment:
this regex needs some documentation to go with it, and is repeated - it should be cached.